### ISSUE TITLE
Some operations with zero-denominator Rationals are broken

### ISSUE THREAD

**Issue by zoffixznet** [rakudo/rakudo#1354]
Filing this so it's not forgotten. `+`, `-` and maybe more ops perform regular numerical math with zero-denominator ops, leading to issues like this one, for example:

```
<Zoffix__> m: ([+] (Inf, Inf, -Inf)).Num.say
<camelia> rakudo-moar c91bcc2af: OUTPUT: «NaN␤»    # CORRECT
<Zoffix__> m: ([+] (Inf, Inf, -Inf)».Rat).Num.say
<camelia> rakudo-moar c91bcc2af: OUTPUT: «Inf␤»       # WRONG
```

I think the fix might simply fall out automatically from [Great Rationals Refactor](https://irclog.perlgeek.de/perl6-dev/2017-07-03#i_14820738) when [`.REDUCE-ME` is tossed]([RT#130774](https://rt.perl.org/Ticket/Display.html?id=130774#ticket-history))

~~`<Zoffix> Maybe try always normalizing all zero-denominator Rats into <1/0>, <-1/0> and <0/0>`~~ that didn't fully work

--------------------------------

Another case:

```

20:50 | lizmat | m: my $a = 0; my $b = 42 / $a; dd $b, $b + $b # not sure this is worth an issue, but it feels wrong
-- | -- | --
20:50 | camelia | rakudo-moar 7b0c5364f: OUTPUT: «Rat $b = <1/0>␤<0/0>␤»

```

Basically all the math stuff needs to be checked and fixed

**Comment by dogbert17** [comment]
At least the original example seems to have been fixed:

    dogbert@dogbert-VirtualBox ~/repos/rakudo $ ./perl6 -v
    This is Rakudo version 2018.10-57-gbfe2ad6 built on MoarVM version 2018.10-26-g318e1ec
    implementing Perl 6.d.
    dogbert@dogbert-VirtualBox ~/repos/rakudo $ ./perl6 -e '([+] (Inf, Inf, -Inf)».Rat).Num.say'
    NaN

**Comment by lizmat** [comment]
Appears to be fixed, also in RakuAST
